src/authenticator.rs

```diff
@@ -335,15 +345,19 @@ mod tests {
     fn flow() {
         use serde_json as json;
 
-        let secret = json::from_str::<ConsoleApplicationSecret>(SECRET).unwrap().installed.unwrap();
-        let res = Authenticator::new(&secret, DefaultAuthenticatorDelegate,
-                        hyper::Client::with_connector(<MockGoogleAuth as Default>::default()),
-                        <MemoryStorage as Default>::default(), None)
-                        .token(&["https://www.googleapis.com/auth/youtube.upload"]);
+        hyper::rt::run(hyper::rt::lazy(|| {
```

Comment:
We need to embed (with option of overriding) a runtime inside the Authenticator so we do  not require the calling application to be inside a Tokio runtime.

In general this move by Hyper to require all applications to have there main inside a Tokio future with no return type seems like a bad design. I am sure I must be missing something. I hope to embed a runtime side Authenticator and then use that.